### ISSUE TITLE
hazelcast-download.properties inside the lib directory

### ISSUE THREAD

**Issue by jerrinot** [hazelcast/hazelcast-jet#2767]
when I build the latest master and unzip the tar.gz distribution then the `lib` directory contains `hazelcast-download.properties`. Is this expected? I guess it should not be there.

**Comment by frant-hartm** [comment]
Yes, it should be there to distinguish call homes between Jet jar from maven, from tar/zip distribution and from docker.

``` 
$ cat hazelcast-jet-4.4-SNAPSHOT/lib/hazelcast-download.properties 
hazelcastDownloadId=distribution

$ docker run hazelcast/hazelcast-jet:latest-snapshot cat /opt/hazelcast-jet/lib/hazelcast-download.properties
hazelcastDownloadId=docker
```

**Comment by jerrinot** [comment]
that's a weird location, isnt it? as a user I am confused by this.

**Comment by olukas** [comment]
Moreover can we add single line comment into the file which explains reason of this file?

**Comment by frant-hartm** [comment]
Options discussed:
- have the file with the correct value in `hazelcast-jet` jar - I don't like this as we would have 3 different jars - one in maven central, one in tar/zip, one in docker
- move the file to `config/` and put the whole config folder on CP - @jerrinot doesn't like the whole folder on CP, single file can't put on CP
- package the file into a jar and put the jar in `lib/`, could be done using yet another assembly

I think the best solution would be to add support for passing a value via a system property to the PhoneHome.

**Comment by cangencer** [comment]
@frant-hartm seems this is how it was done for IMDG: https://github.com/hazelcast/hazelcast-docker/pull/230/files#diff-fcaf45f62a2fdc483b0edd5506e0bb6a7fbd11bba72dedcef1ab36117aecda20R56
